src/coreComponents/finiteVolume/TwoPointFluxApproximation.cpp

```diff
@@ -594,6 +594,154 @@ void TwoPointFluxApproximation::addToFractureStencil( MeshLevel & mesh,
+
+      stackArray1d< R1Tensor, maxElems > stencilCellCenterToEdgeCenters( numElems );
+      stackArray1d< integer, maxElems > isGhostConnectors( numElems );
+
+      //TODO get edge geometry
+      R1Tensor const edgeCenter = embSurfEdgeManager->calculateCenter( ke, X );
+      real64 const edgeLength   = embSurfEdgeManager->calculateLength( ke, X ).L2_Norm();
+
+      // loop over all embedded surface elements attached to the connector and add them to the stencil
+      for( localIndex kes = 0; kes < numElems; kes++ )
+      {
+        localIndex const fractureElementIndex = edgeToEmbSurfacesMap[ke][kes];
+
+        // compute distance between cell centers
+        real64 cellCenterToEdgeCenter[ 3 ];
+        LvArray::tensorOps::copy< 3 >( cellCenterToEdgeCenter, edgeCenter );
+        LvArray::tensorOps::subtract< 3 >( cellCenterToEdgeCenter, fractureElemCenter[fractureElementIndex] );
+
+        // form the CellStencil entry
+        stencilCellsRegionIndex[kes]    = fractureRegionIndex;
+        stencilCellsSubRegionIndex[kes] = 0;  // there is only one subregion.
+        stencilCellsIndex[kes]          = fractureElementIndex;
+
+        //TODO use the proper geometrical info to compute the weight.
+        stencilWeights[kes] =  1.0 / 12.0 * edgeLength / LvArray::tensorOps::l2Norm< 3 >( cellCenterToEdgeCenter );
+
+        LvArray::tensorOps::copy< 3 >( stencilCellCenterToEdgeCenters[kes], cellCenterToEdgeCenter );
+      }
+
+      // add/overwrite the stencil for index fci
+      fractureStencil.add( numElems,
+                           stencilCellsRegionIndex.data(),
+                           stencilCellsSubRegionIndex.data(),
+                           stencilCellsIndex.data(),
+                           stencilWeights.data(),
+                           connectorIndex );
+
+      fractureStencil.add( numElems,
+                           stencilCellCenterToEdgeCenters.data(),
+                           connectorIndex );
+
+      connectorIndex++;
+    }
+  }
+
+  // Add connections EmbeddedSurface to/from CellElements.
+  arrayView1d< localIndex const > const & elemRegionList    = fractureSubRegion->getSurfaceToRegionList();
+  arrayView1d< localIndex const > const & elemSubRegionList = fractureSubRegion->getSurfaceToSubRegionList();
+  arrayView1d< localIndex const > const & elemList          = fractureSubRegion->getSurfaceToCellList();
+  arrayView1d< real64 const >     const & ConnectivityIndex = fractureSubRegion->getConnectivityIndex();
+
+  ElementRegionManager::ElementViewAccessor< arrayView1d< R1Tensor const > > const permeabilityTensor =
+    elemManager->ConstructArrayViewAccessor< R1Tensor, 1 >( m_coeffName );
+
+  // start from last connectorIndex from cell-To-cell connections
+  connectorIndex = cellStencil.size();
+
+  // loop over the embedded surfaces and add connections to cellStencil
+  for( localIndex kes=0; kes  < fractureSubRegion->size(); kes++ )
+  {
+    //if( ghostRank[kes] < 0 )
```

Comment:
I am honestly not sure.

Reply:
It seems that `EmbeddedSurfaceSubRegion::inheritGhostRank()` sets the ghosting of embedded surface elements to be identical to the containing volume element. In that case, there is no reason to create a stencil entry between two ghosted elements since any matrix/rhs contribution would have to be discarded anyway. So I'd say you could reinstate the `ghostRank` check here.